### ISSUE TITLE
MySQL fails to start (or to setup socket) every couple of times

### ISSUE THREAD

**Issue by arvenil** [travis-ci/travis-ci#7790]
https://travis-ci.org/percona/qan-agent/builds/234293651

1 build, 3 jobs, jobs only differ version of golang

I run into issue where my builds where failing basically most of the times because some of the jobs were sometimes failing. Like in the above example, 1st job passed, but next 2 didn't. If I restart job it probably gonna pass.

Jobs were failing at interaction with MySQL, so I thought that maybe I start jobs too early, before MySQL actually accepts connections. So I've introduced "wait for MySQL" script, the script runs every second a query and so checks if MySQL is online. So now, randomly, my jobs pass or either... they hang forever, because it looks like MySQL sometimes doesn't start at all.

```yaml
dist: trusty

language: go

go:
  - 1.6.x
  - 1.7.x
  - 1.8.x

services:
  - mysql

env:
  global:
    - MYSQL_USER="root"
    - MYSQL_PASS="travis"
    - MYSQL_HOST="localhost"
    - MYSQL_PORT="3306"
    - PCT_TEST_MYSQL_DSN="root:travis@tcp(localhost:3306)/?parseTime=true"

addons:
  apt:
    sources:
      - sourceline: 'deb http://repo.percona.com/apt precise testing'
        key_url: 'http://keyserver.ubuntu.com/pks/lookup?op=get&fingerprint=on&search=0x8507EFA5'
    packages:
      - percona-toolkit

install:
  - go get -u github.com/Masterminds/glide
  - glide install

before_script:
  - |
    until mysql -u${MYSQL_USER} -p${MYSQL_PASS} -h ${MYSQL_HOST} -P ${MYSQL_PORT} -e 'SELECT 1' > /dev/null; do
      >&2 echo "MySQL ${MYSQL_HOST}:${MYSQL_PORT} for user ${MYSQL_USER} is unavailable - sleeping"
      sleep 1
    done
    >&2 echo "MySQL ${MYSQL_HOST}:${MYSQL_PORT} is up"

script:
  - test/runner.sh

notifications:
  email:
    on_success: change
    on_failure: change
```

**Comment by BanzaiMan** [comment]
My guess is that containers may not have enough resources to start MySQL in some cases. If you set `sudo: required`, is it any more reliable?

**Comment by arvenil** [comment]
I just dropped `dist: trusty`, so we will see... I just made 3 runs in a row (so 9 jobs) and they came all 100% ok.

I've opened this issue, because to be honest it looked really weird to me that MySQL for container solution fails so often... I know it's beta, but still this happens way to often. It happens so often that it made me think it's somehow fault on my side. Maybe it would be worth to monitor this on travis side to give you an idea what's the scale of the issue. Or if anybody has an idea what I'm doing wrong then let me know ;)
